iothub/device/src/Transport/Amqp/AmqpAuthenticationRefresher.cs

```diff
@@ -58,21 +41,21 @@ public async Task InitLoopAsync(TimeSpan timeout)
                     ConnectionString,
                     ConnectionString.AmqpEndpoint,
                     Audience,
-                    Audience,
-                    AccessRightsHelper.AccessRightsToStringArray(AccessRights.DeviceConnect),
-                    timeoutHelper.RemainingTime()
+                    WebUtility.UrlEncode(Audience),
+                    AccessRightsStringArray,
+                    timeout
                 ).ConfigureAwait(false);
             if (expiry < DateTime.MaxValue)
             {
                 StartLoop(expiry, newToken);
             }
-            if (Logging.IsEnabled) Logging.Exit(this, timeoutHelper.RemainingTime(), $"{nameof(InitLoopAsync)}");
+            if (Logging.IsEnabled) Logging.Exit(this, timeout, $"{nameof(InitLoopAsync)}");
         }
 
         private void StartLoop(DateTime expiry, CancellationToken cancellationToken)
         {
             if (Logging.IsEnabled) Logging.Enter(this, expiry, $"{nameof(StartLoop)}");
-            RefreshLoop = RefreshLoopAsync(expiry, cancellationToken);
+            RefreshLoop = RefreshLoopAsync(expiry, cancellationToken).ConfigureAwait(false);
```

Comment:
ConfigureAwait is wrong here. Task needs to be assigned only.

Reply:
If not, it will fail token refresh e2etest.